main.tf

```
@@ -1,47 +1,55 @@
 locals {
-  # Define resource types that support cold storage
+  # Define key resource types that support cold storage
   supported_resource_types = [
-    "arn:aws:cloudformation:*:*:stack/*",
-    "arn:aws:dynamodb:*:*:table/*",
-    "arn:aws:ec2:*:*:volume/*",
-    "arn:aws:elasticfilesystem:*:*:file-system/*",
-    "arn:aws:ec2:*:*:instance/*", # For SAP HANA on EC2
-    "arn:aws:timestream:*:*:database/*",
-    "arn:aws:backup:*:*:backup-vault/*",
-    "arn:aws:backup:*:*:recovery-point/*"
+    "cloudformation",
+    "dynamodb",
+    "ec2",
+    "elasticfilesystem",
+    "timestream",
+    "backup"
   ]
 
   # Determine the SNS topic ARN to use
   sns_topic_arn = var.sns_topic_arn == "" ? aws_sns_topic.bkup_sns_topic[0].arn : var.sns_topic_arn
 
   # Identify resources that don't support cold storage
-  cold_storage_unsupported_resources = [
+  cold_storage_unsupported_resources = var.enable_cold_storage_check ? [
     for arn in var.source_arns : arn
-    if !can(regex(join("|", local.supported_resource_types), arn))
-  ]
-
-  # Error message for resources that don't support cold storage
-  cold_storage_error_message = "Error: Cold storage is not supported for the following resources: ${join(", ", local.cold_storage_unsupported_resources)}."
+    if !can(contains(local.supported_resource_types, split(":", arn)[2]))
```

```suggestion
    if !contains(local.supported_resource_types, split(":", arn)[2])
```
I think the `can` is not correct here since `contains` already returns a boolean value.

---

Sure, @briskt  the reason i went with  the "can" over there is to handle the format error of the arn just to in a safe side.

---

Not bad to be safe! I wasn't even aware of the `can` function so I tested it in a small test file on my machine. It seemed to interfere with correct operation. I suppose it would have to be done differently to work correctly.